src/components/OfflineSearch.ts

```diff
@@ -0,0 +1,174 @@
+                    className: `btn-transparent ${this.state.buttonVisibility}`,
+                    ref: button => this.searchButton = button as HTMLButtonElement
+                },
+                    createElement("span", { className: "glyphicon glyphicon-remove" })
+                )
+            )
+            : createElement("div", { className: "search-container-hidden" })
+        ));
+    }
+
+    componentDidMount() {
+        if (this.props.showSearchBar) {
+          this.findListView();
+          if (this.isValidWidget(this.targetWidget)) {
+            this.setUpEvents();
+            this.setDefaultSearch();
+          }
+        }
+    }
+
+    componentWillUnmount() {
+        if (this.props.showSearchBar) {
+            this.searchButton.removeEventListener("click", this.onClear);
+            this.searchInput.removeEventListener("keyup", this.onSearchKeyDown);
+        }
+    }
+
+    private findListView() {
+        let queryNode = findDOMNode(this).parentNode as HTMLElement;
+        let targetNode: HTMLElement | null = null;
+
+        while (!targetNode) {
+            targetNode = queryNode.querySelector(`.mx-name-${this.props.targetGridName}`) as HTMLElement;
+            if (window.document.isEqualNode(queryNode)) break;
+            queryNode = queryNode.parentNode as HTMLElement;
+        }
+
+        if (!targetNode) {
+            this.setState({ alertMessage: `search offline widget: unable to find grid with the name "${this.props.targetGridName}"` });
+        }
+
+        this.targetWidget = targetNode ? dijitRegistry.byNode(targetNode) : null;
+    }
+
+    private isValidWidget(targetWidget: ListView): boolean {
+        if (targetWidget && targetWidget.declaredClass === "mxui.widget.ListView") {
+            if (
+                targetWidget._onLoad &&
+                targetWidget._loadMore &&
+                targetWidget._renderData &&
+                targetWidget._datasource &&
+                targetWidget._datasource.atEnd &&
+                typeof targetWidget._datasource._pageSize !== "undefined" &&
+                typeof targetWidget._datasource._setSize !== "undefined"
+            ) {
+                return true;
+            } else {
+                this.setState({ alertMessage: "search offline widget: this Mendix version is incompatible with the offline search widget" });
+            }
+        } else {
+            this.setState({ alertMessage: `search offline widget: supplied target name "${this.props.targetGridName}" is not of the type listview` });
+        }
+
+        return false;
+    }
+
+    private setUpEvents() {
+        this.searchButton.addEventListener("click", this.onClear);
+        this.searchInput.addEventListener("keyup", this.onSearchKeyDown);
+    }
+
+    private setDefaultSearch() {
+        this.searchInput.value = this.props.defaultQuery;
+        this.updateConstraints(this);
+    }
+
+    private onSearchKeyDown(event: CustomEvent) {
```

Comment:
`event` parameter is not being used. Please remove. Activated `tsconfig` for `noUnusedParameters`